### PR TITLE
Keycloak Refactor

### PR DESCRIPTION
**Pull request by FuriousLlama**
Refactored keycloak sync to use the keycloak repository used by the aPI. 

ALso removed tools.Core and removed redundant models

### CODE REVIEW

File: source/backend/keycloak/IKeycloakService.cs
Comment (FuriousLlama) [inline source/backend/keycloak/IKeycloakService.cs:1]: This was actually renamed to IKeycloackRepository.cs same with the implementation file

File: source/backend/keycloak/Models/RoleModel.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ public class RoleModel
         /// <summary>
         /// get/set - whether or not this role is a composite role.
         /// </summary>
-        public bool Composite { get; set; }
+        public bool? Composite { get; set; }
```
Comment (devinleighsmith) [inline source/backend/keycloak/Models/RoleModel.cs:18]: Was this incorrect in the current implementation?

File: tools/keycloak/sync/Syncronizer/SyncFactory.cs [hunk 2]
```diff
@@ -425,12 +425,12 @@ private async Task<KModel.UserModel> GetUserAsync(PModel.UserModel user)
             try
             {
                 // Make a request to keycloak to find a matching user.
-                var response = await _keycloakManagementClient.HandleRequestAsync<ResponseWrapperModel<IEnumerable<KModel.UserModel>>>(HttpMethod.Get, $"{_keycloakManagementClient.GetEnvUri()}/idir/users?guid={user.GuidIdentifierValue.ToString().Replace("-", string.Empty)}");
-                if (response.Data.Count() > 1)
+                var users = await _keycloakService.GetUsersAsync(user.GuidIdentifierValue);
```
Comment (devinleighsmith) [inline tools/keycloak/sync/Syncronizer/SyncFactory.cs:428]: this all looks good, using the repository should ensure consistency across the two projects.

**Comment by devinleighsmith** [comment]
@FuriousLlama this doesn't compile either, am I doing something wrong?

**Review by devinleighsmith — APPROVED** [review]

**Pull request merged**